### PR TITLE
fix #293 support a list of rpc_endpoint for collecting the metadata

### LINKED ISSUE #293: Support a list of `rpc_endpoint` for collecting the metadata
Support a list of `rpc_endpoint` for collecting the metadata.

**The problem**
Some of RPC nodes are not available when updating the metadata for networks. In that case the update process fails.

**The solution**
1. Add a list of `rpc_endpoint` to the `config.toml` file.
2. During the update if the 1st RPC node is not available check 2nd, 3rd, etc. nodes.
3. If all RPC nodes are not available then fail.

### PR DESCRIPTION
**Pull request by prybalko**
Make 2 available option: 
**rpc_endpoint** is a string
```
[[chains]]
rpc_endpoint = "wss://rpc.polkadot.io"
```
OR 
**rpc_endpoints** is a list of strings
```
[[chains]]
rpc_endpoints = ["wss://rpc.polkadot.io"]
```

### CODE REVIEW

File: cli/src/collector/export.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub(crate) fn export_specs(config: &AppConfig, fetcher: impl Fetcher) -> Result<
             ExportChainSpec {
                 title: chain.title.as_ref().unwrap_or(&chain.name).clone(),
                 color: chain.color.clone(),
-                rpc_endpoint: chain.rpc_endpoint.clone(),
+                rpc_endpoint: chain.rpc_endpoints[0].clone(), // keep only the first one
```
Comment (pgolovkin) [inline cli/src/collector/export.rs:43]: Is it possible to get the node that was used for getting the chainspec?

**Review by pgolovkin — CHANGES_REQUESTED** [review]

File: cli/src/collector/export.rs (hunk 1 above)
Comment (prybalko) [reply cli/src/collector/export.rs:43]: It is possible, although I'm not sure how useful that is. 
I see having multiple endpoints as a failover option. That means there is an "official" endpoint and some others for backup\failover\load balancing. Not sure if users are interested in knowing the details. 
Moreover, chainspecs are fetched with every `make collector`, so we might get different endpoints after each deployment. That might be confusing.

Comment (pgolovkin) [reply cli/src/collector/export.rs:43]: Got it. It's a very good point!

**Review by pgolovkin — APPROVED** [review]

**Comment by pgolovkin** [comment]
The example output
```
[2022-08-12T11:32:16Z INFO ] ⚙️  Generating unsigned_statemine_specs.png...
[2022-08-12T11:32:19Z INFO ] ⚙️  Generating unsigned_statemine_metadata_9230.apng...

[2022-08-12T11:34:12Z WARN ] Failed to fetch wss://turing.api.onfinality.io/public-ws: Failed { url: "wss://turing.api.onfinality.io/public-ws", error: "Networking or low-level protocol error: Connection rejected with status code: 200" }
getting title
[2022-08-12T11:34:14Z INFO ] ⚙️  Generating unsigned_turing_specs.png...
[2022-08-12T11:36:15Z WARN ] Failed to fetch wss://turing.api.onfinality.io/public-ws: Failed { url: "wss://turing.api.onfinality.io/public-ws", error: "Networking or low-level protocol error: Error in the WebSocket handshake: i/o error: unexpected end of file" }
[2022-08-12T11:36:17Z INFO ] ⚙️  Generating unsigned_turing_metadata_285.apng...
```

**Pull request merged**